src/libutils/oglapphelpers/glsl.cpp

```diff
@@ -119,7 +121,7 @@ void AllocateTexture2D(unsigned index, unsigned & texId,
 
     glActiveTexture(GL_TEXTURE0 + index);
 
-    if (height > 1)
+    if (height > 1 || !allowTexture1D)
```

Comment:
Flipping this conditional around to `height == 1 && allowTexture1D` would make it clearer.